src/app/account-details/account-details.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { AngularFireDatabase } from 'angularfire2/database';
+import * as firebase from 'firebase';
+import {AngularFireAuth} from 'angularfire2/auth';
+import { UserService } from '../core/user.service';
+import { AuthService } from '../core/auth.service';
+import { FirebaseUserModel } from '../core/user.model';
+import { ActivatedRoute } from '@angular/router';
+import { AngularFireStorage } from 'angularfire2/storage';
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+
+@Component({
+  selector: 'app-account-details',
+  templateUrl: './account-details.component.html',
+  styleUrls: ['./account-details.component.scss']
+})
+export class AccountDetailsComponent implements OnInit {
+  user: FirebaseUserModel ;
+  userDetails: any;
+  imageBack: string;
+  imageFront: string;
+  constructor(
+    public matDialog: MatDialog,
+    public userService: UserService,
+    public authService: AuthService,
+    private route: ActivatedRoute,
+    private db: AngularFireDatabase,
+    private storage: AngularFireStorage,
+  ) { }
+
+  ngOnInit() {
+    this.route.data.subscribe(routeData => {
+      const data = routeData['data'];
+      if (data) {
```

Comment:
this IF is redundant, you can be sure you get this data. If data is not provided, then you already handled this in resolver, by navigating user back to login (this.router.navigate(['/login'] on error)